features/fixtures/mazerunner/app/src/main/java/com/bugsnag/android/mazerunner/MainActivity.kt

```diff
@@ -73,6 +76,33 @@ class MainActivity : Activity() {
         log("MainActivity.onResume complete")
     }
 
+    private fun setMazeRunnerAddress() {
+        val context = MazerunnerApp.applicationContext()
+        val externalFilesDir = context.getExternalFilesDir(null)
+        val configFile = File(externalFilesDir, "fixture_config.json")
+        log("Attempting to read Maze Runner address from config file ${configFile.path}")
+
+        // Poll for the fixture config file
+        val pollStart = System.currentTimeMillis()
+        while (System.currentTimeMillis() - pollStart < CONFIG_FILE_TIMEOUT) {
+            if (configFile.exists()) {
+                val fileContents = configFile.inputStream().use { it.reader().readText() }
```

Comment:
Wrapping an `InputStream` in a `Reader` negates the need for `use` here, as `InputStreamReader.close()` will close the underlying `InputStream`.

```suggestion
                val fileContents = configFile.inputStream().reader().readText()
```

Reply:
Not sure I follow - would I then need to close the steam explicitly? Could this be simplified even further?
```suggestion
                val fileContents = configFile.readText()
```

Reply:
`use` is shorthand for "close this when I'm done with it" - I assumed that is what you meant there.

Yes, you could 👍  I forgot they added that to `File` as well 😆